test/suites/zombie_tanssi_relay_eth_bridge/test_zombie_tanssi_relay_eth_bridge.ts

```diff
@@ -295,6 +325,86 @@ describeSuite({
+
+                const currentBlock = (await relayApi.rpc.chain.getBlock()).block.header.number.toNumber();
+                const blockToFetchRewardEventFrom = currentBlock + blocksToWaitForRewards;
+                const blockToFetchSlashEventFrom = currentBlock + blocksToWaitForSlashes;
+
+                console.log("We will wait till", currentBlock + blocksToWaitFor, "blocks");
+
+                await context.waitBlock(blocksToWaitFor, "Tanssi-relay");
+                const relayApiAtRewardEventBlock = await relayApi.at(
+                    await relayApi.query.system.blockHash(blockToFetchRewardEventFrom)
+                );
+                const relayApiAtSlashEventBlock = await relayApi.at(
+                    await relayApi.query.system.blockHash(blockToFetchSlashEventFrom)
+                );
+
+                // Get the reward event
+                const rewardBlockEvents = await relayApiAtRewardEventBlock.query.system.events();
+                const filteredEventsForReward = rewardBlockEvents.filter((a) => {
+                    return a.event.method == "RewardsMessageSent";
+                });
+                expect(filteredEventsForReward.length).to.be.equal(1);
+                const rewardEvent = filteredEventsForReward[0];
+                // Extract message id
+                const rewardMessageId = rewardEvent.event.toJSON().data[0];
+                console.log("Reward message id:", rewardMessageId);
+
+                // Get the slash event
+                const slashBlockEvents = await relayApiAtSlashEventBlock.query.system.events();
+                const filteredEventsForSlash = slashBlockEvents.filter((a) => {
+                    return a.event.method == "SlashesMessageSent";
+                });
+                expect(filteredEventsForSlash.length).to.be.equal(1);
+                const slashEvent = filteredEventsForSlash[0];
+                // Extract message id
+                const slashMessageId = slashEvent.event.toJSON().data[0];
+                console.log("Slash message id:", slashMessageId);
+
+                let rewardMessageReceived = false;
+                let slashMessageReceived = false;
+                let rewardMessageSuccess = false;
+                let slashMessageSuccess = false;
+
+                gatewayContract.on("InboundMessageDispatched", (_channelID, _nonce, messageID, success) => {
+                    if (rewardMessageId == messageID) {
+                        rewardMessageReceived = true;
+                        rewardMessageSuccess = success;
+                    } else if (slashMessageId == messageID) {
+                        slashMessageReceived = true;
+                        slashMessageSuccess = success;
+                    }
+                });
+
+                while (!rewardMessageReceived || !slashMessageReceived) {
+                    await sleep(1000);
+                }
```

Comment:
This is an infinite loop or I don't know javascript

Reply:
until the test times out right?

Reply:
which right now is set to 10 minutes since it starts

Reply:
maybe we should increase the timeout to 15, I think we are soon going to be on the limit

Reply:
![image](https://github.com/user-attachments/assets/6de3ae23-dc5c-4f48-931f-46dd85effe0d)
Actually we should have timed out, anybody knows why we did not?

Reply:
I believe it's 10 minutes per test case yes, because zombie_data_preservers took 40 minutes to timeout.

Reply:
ok then I think it's ok